pennylane_lightning_gpu/src/simulator/StateVectorCudaRaw.hpp

```diff
@@ -0,0 +1,1098 @@
+              {"RZ",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyRZ(std::forward<decltype(wires)>(wires),
+                           std::forward<decltype(adjoint)>(adjoint),
+                           std::forward<decltype(params[0])>(params[0]));
+               }},
+              {"PhaseShift",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyPhaseShift(
+                       std::forward<decltype(wires)>(wires),
+                       std::forward<decltype(adjoint)>(adjoint),
+                       std::forward<decltype(params[0])>(params[0]));
+               }},
+              {"CRX",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyCRX(std::forward<decltype(wires)>(wires),
+                            std::forward<decltype(adjoint)>(adjoint),
+                            std::forward<decltype(params[0])>(params[0]));
+               }},
+              {"CRY",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyCRY(std::forward<decltype(wires)>(wires),
+                            std::forward<decltype(adjoint)>(adjoint),
+                            std::forward<decltype(params[0])>(params[0]));
+               }},
+              {"CRZ",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyCRZ(std::forward<decltype(wires)>(wires),
+                            std::forward<decltype(adjoint)>(adjoint),
+                            std::forward<decltype(params[0])>(params[0]));
+               }},
+              {"ControlledPhaseShift",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyControlledPhaseShift(
+                       std::forward<decltype(wires)>(wires),
+                       std::forward<decltype(adjoint)>(adjoint),
+                       std::forward<decltype(params[0])>(params[0]));
+               }},
+              {"Rot",
+               [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyRot(std::forward<decltype(wires)>(wires),
+                            std::forward<decltype(adjoint)>(adjoint),
+                            std::forward<decltype(params)>(params));
+               }},
+              {"CRot", [&](auto &&wires, auto &&adjoint, auto &&params) {
+                   applyCRot(std::forward<decltype(wires)>(wires),
+                             std::forward<decltype(adjoint)>(adjoint),
+                             std::forward<decltype(params)>(params));
+               }}} {
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecCreate(
+            /* custatevecHandle_t* */ &handle));
+    };
+
+    StateVectorCudaRaw(void *gpu_data, size_t length, cudaStream_t stream = 0)
+        : StateVectorCudaRaw(Util::log2(length), stream) {
+        CFP_t *ptr = BaseType::getData();
+        ptr = reinterpret_cast<CFP_t *>(gpu_data);
```

Comment:
What does this line do? (as `ptr` doesn't seem to be used)

Reply:
I believe this is an error that I made on previous tests with casting rules --- completely skipped my brain, thanks for this.